app/controllers/assignments_controller.rb

```diff
@@ -0,0 +1,149 @@
+class AssignmentsController < ApplicationController
+
+  def index
+
+    @taught_class = TaughtClass.find(params[:taught_class_id])
+    @assignments_set = policy_scope(Assignment).where(taught_class: @taught_class)
+
+  end
+
+
+  def new
+    @taught_class = TaughtClass.find(params[:taught_class_id])
+    @assignment = Assignment.new
+    @assignment.taught_class = @taught_class
+
+    authorize @assignment
+
+  end
+
+
+  def create
+    @taught_class = TaughtClass.find(params[:taught_class_id])
+    @assignment = Assignment.new(assignment_params)
+    @assignment.taught_class = @taught_class
+    authorize @assignment
+
+
+    if @assignment.save
+      redirect_to assignment_path(@assignment)
+    else
+      puts @assignment.errors.full_messages
+      render :new
+    end
+
+  end
+
+  def edit
+    @assignment = Assignment.find(params[:id])
+    authorize @assignment
+
+  end
+
+  def update
+    @assignment = Assignment.find(params[:id])
+    authorize @assignment
```

Comment:
I'm not completely sure about puntdit, I use cancancan. But maybe this is good?

I would maybe DRY it up a little